app/models/live_result.rb

```diff
@@ -41,7 +41,7 @@ def to_solve_time(field)
 
   def potential_score
     rank_by = round.format.sort_by == 'single' ? 'best' : 'average'
-    complete? ? self[rank_by.to_sym] : BEST_POSSIBLE_SCORE
+    complete? ? to_solve_time(rank_by) : SolveTime.new(event_id, rank_by, BEST_POSSIBLE_SCORE)
```

Comment:
This changes the return value of the method entirely. I assume this is intended, but why?